### PR TITLE
feat #2218: Inject platform environment variables

### LINKED ISSUE #2218: Inject Platform Environment variables
See https://github.bus.zalan.do/SRE/SRE-program/issues/433

### PR DESCRIPTION
**Pull request by zeitlinger**
Fixes #2218.

### CODE REVIEW

File: cluster/manifests/admission-control/teapot.yaml [hunk 1]
```diff
@@ -14,7 +14,7 @@ webhooks:
       - operations: [ "CREATE" ]
         apiGroups: [""]
         apiVersions: ["v1"]
-        resources: ["pods"]
+        resources: ["pods", "pods/binding"]
```
Comment (aermakov-zalando) [inline cluster/manifests/admission-control/teapot.yaml]: Please split it into a separate rule, with `failurePolicy: Ignore`.

Comment (zeitlinger) [reply cluster/manifests/admission-control/teapot.yaml]: good idea - does this mean we could propagate the errors from binding (again)?
Might make troubleshooting easier.

Comment (zeitlinger) [reply cluster/manifests/admission-control/teapot.yaml]: anyways, I've changed as advised.

File: cluster/manifests/admission-control/teapot.yaml [hunk 2]
```diff
@@ -15,6 +15,16 @@ webhooks:
         apiGroups: [""]
         apiVersions: ["v1"]
         resources: ["pods"]
+  - name: pod-binding-admitter.teapot.zalan.do
```
Comment (aermakov-zalando) [inline cluster/manifests/admission-control/teapot.yaml:19]: This should be present only when `teapot_admission_controller_inject_environment_variables` is enabled.

File: test/e2e/admission_controller.go [hunk 3]
```diff
@@ -37,6 +37,8 @@ const (
 	deploymentId = "d-2ajbkvmtqo5isznh3m2raj2bdp"
 	pipelineId   = "l-2kwqgqevuqwsje5a9ukhsk4bdd"
 	application  = "e2e-test-application"
+	component  = "e2e-test-component"
```
Comment (aermakov-zalando) [inline test/e2e/admission_controller.go]: `gofmt`?

**Comment by zeitlinger** [comment]
:+1:

**Comment by aermakov-zalando** [comment]
👍

File: cluster/node-pools/master-default/userdata.yaml [hunk 4]
```diff
@@ -210,6 +211,17 @@ write_files:
...
 {{- end }}
+{{- if eq .Cluster.ConfigItems.teapot_admission_controller_inject_environment_variables "true" }}
+            - --pod-enable-env-inject
+            - --pod-inject-env=_PLATFORM_ACCOUNT={{ .Cluster.Alias }}
+            - --pod-inject-env=_PLATFORM_OPENTRACING_TAG_ACCOUNT={{ .Cluster.Alias }}
+            - --pod-inject-env=_PLATFORM_OPENTRACING_LIGHTSTEP_COLLECTOR_PORT=8443
+            - --pod-inject-env=_PLATFORM_OPENTRACING_LIGHTSTEP_COLLECTOR_HOST=tracing.stups.zalan.do
+            - --pod-inject-env=_PLATFORM_OPENTRACING_LIGHTSTEP_ACCESS_TOKEN={{ .ConfigItems.lightstep_token }}
+{{- if eq .Cluster.Environment "e2e" }}
+              - --pod-inject-env=_PLATFORM_E2E=injected
```
Comment (mikkeloscar) [inline cluster/node-pools/master-default/userdata.yaml]: indentation looks wrong here

Comment (zeitlinger) [reply cluster/node-pools/master-default/userdata.yaml]: fixed - strangely it hasn't let to an error

**Comment by aermakov-zalando** [comment]
👍

**Comment by zeitlinger** [comment]
:+1:

**Comment by mikkeloscar** [comment]
:+1:

**Pull request merged**